### PR TITLE
Add a jenkins job to run a rake task:

### PR DESCRIPTION
**Pull request by ssatish**
That will take content from yaml files in email-campaign-frontend repo and
move it to draft content store using publishing api

The job is configured to run after every push in the repo

### CODE REVIEW

**Comment by surminus** [comment]
`modules/govuk_jenkins/templates/jobs/departmental_projects_content_change_to_draft.yaml` should be renamed to `modules/govuk_jenkins/templates/jobs/departmental_projects_content_change_to_draft.yaml.erb`

File: modules/govuk_jenkins/templates/jobs/departmental_projects_content_change_to_draft.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+- scm:
+    name: departmental_projects_content_change_to_draft
```
Comment (surminus) [inline modules/govuk_jenkins/templates/jobs/departmental_projects_content_change_to_draft.yaml]: I believe this name needs to be different from the job name - check the other templates as examples.

File: modules/govuk_jenkins/templates/jobs/departmental_projects_content_change_to_draft.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    display-name: Departmental_projects_content_change_to_draft
+    project-type: freestyle
+    description: "<p>Supports github publishing by running rake task to put changed content to draft content store</p>"
+    scm:
+      - departmental_projects_content_change_to_draft
+    logrotate:
+      artifactNumToKeep: 30
+    builders:
+        - shell: |
+            ssh deploy@email-campaign-frontend-1.frontend.integration "cd /var/apps/email-campaign-frontend && 	"
```
Comment (surminus) [inline modules/govuk_jenkins/templates/jobs/departmental_projects_content_change_to_draft.yaml]: This line doesn't seem correct.

**Comment by ssatish** [comment]
Changed the names and put in the command.

**Comment by ssatish** [comment]
I have updated this PR to incorporate our new simplified strategy to editing content. We will now draft and publish all in one stroke. This strategy has been approved by Dai.

File: modules/govuk_jenkins/templates/jobs/departmental_projects_publish_content_change.yaml.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    display-name: Departmental_projects_publish_content_change
+    project-type: freestyle
+    description: "<p>Supports github publishing by running rake tasks to publish changed content</p>"
+    scm:
+      - dp_publish_content_change
+    logrotate:
+      artifactNumToKeep: 30
+    builders:
+        - shell: |
+            ssh deploy@email-campaign-frontend-1.frontend.integration "cd /var/apps/email-campaign-frontend && govuk_setenv email-campaign-frontend bundle exec rake publish_content:draft_and_publish_all"
```
Comment (rjw1) [inline modules/govuk_jenkins/templates/jobs/departmental_projects_publish_content_change.yaml.erb]: You probably dont want to hard code integration. `email-campaign-frontend-1.frontend` should work in all envs as a hostname

File: modules/govuk_jenkins/templates/jobs/departmental_projects_publish_content_change.yaml.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    triggers:
+        - project: email-campaign-frontend
+          condition: 'SUCCESS'
+    wrappers:
+        - ansicolor:
+            colormap: xterm
```
Comment (rjw1) [inline modules/govuk_jenkins/templates/jobs/departmental_projects_publish_content_change.yaml.erb]: Could you add a new line.

File: modules/govuk_jenkins/manifests/job/departmental_projects_publish_content_change.pp [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+#
+# Create a file on disk that can be parsed by jenkins-job-builder
+#
+class govuk_jenkins::job::departmental_projects_publish_content_change {
+  file { '/etc/jenkins_jobs/jobs/departmental_projects_publish_content_change.yaml':
+    ensure  => present,
+    content => template('govuk_jenkins/jobs/departmental_projects_publish_content_change.yaml.erb'),
+    notify  => Exec['jenkins_jobs_update'],
+  }
+}
```
Comment (rjw1) [inline modules/govuk_jenkins/manifests/job/departmental_projects_publish_content_change.pp]: could you add a newline?

**Comment by ssatish** [comment]
@Bob I have changed the hostname so it works on all envs, and added newlines.

File: modules/govuk_jenkins/templates/jobs/departmental_projects_publish_content_change.yaml.erb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    scm:
+      - dp_publish_content_change
+    logrotate:
+      artifactNumToKeep: 30
+    builders:
+        - shell: |
+            ssh deploy@email-campaign-frontend-1.frontend "cd /var/apps/email-campaign-frontend && govuk_setenv email-campaign-frontend bundle exec rake publish_content:draft_and_publish_all"
+    triggers:
+        - project: email-campaign-frontend
+          condition: 'SUCCESS'
```
Comment (alexmuller) [inline modules/govuk_jenkins/templates/jobs/departmental_projects_publish_content_change.yaml.erb]: I don't think that a `project` is a valid type for `triggers`. I got this error when running this YAML locally:

```
jenkins_jobs.errors.JenkinsJobsException: Unknown entry point or macro 'project' for component type: 'trigger'.
```

**Comment by ssatish** [comment]
@alexmuller  No more triggers.

**Comment by alexmuller** [comment]
Looks good to me, thanks!

**Pull request merged**